src/utils/ShuffleGraph.java

```diff
@@ -66,18 +70,50 @@ public void shuffleArray(int[] array, ArrayVisualizer arrayVisualizer) {
     }
 
     public void addDisconnected(ShuffleInfo shuffle) {
-        this.nodes.add(new Node(shuffle, this));
+        this.nodes.add(new ShuffleNode(shuffle, this));
     }
 
     public void addDisconnected(ShuffleInfo shuffle, int x, int y) {
-        this.nodes.add(new Node(shuffle, this, x, y));
+        this.nodes.add(new ShuffleNode(shuffle, this, x, y));
+    }
+
+    public Point findSafeCoordinate(int baseX, int baseY, int offsetX, int offsetY) {
```

Comment:
shouldn't this be "coordinate**s**"?

Reply:
It could be either I think, right?

Reply:
even if, "coordinates" is definitely less ambiguous (since the other can refer to an individual value, like "x-coordinate")

also did a little searching, can't find a case where it's used like this